src/peer.rs

```diff
@@ -197,6 +212,10 @@ pub(crate) fn new_peer<T: HandshakeHandler>(
                 }
                 Err(err) => {
                     println!("Peer err {:?}", err);
+                    let mut active_connections = active_connections.write();
+                    active_connections.connections.remove(&peer_id).expect("Unable to remove peer id");
+                    let _ = write_thread_handle.join();
```

Comment:
Will the writer thread stop if the receive fail ? Because if the writer has nothing to write (that could happens especially if the connection isn't in the `active_connections` because it's the source of node connected for handlers) it will never try to write and then fail

Reply:
Hmmm ok, I don't see clearly how everything is wired yet. In which case does this have to fail and panic ?